lib/jxl/modular/transform/palette.h

```diff
@@ -389,6 +389,11 @@ static Status CheckPaletteParams(const Image &image, uint32_t begin_c,
       return false;
     }
   }
+  for (size_t c = begin_c; c <= end_c; c++) {
+    if (image.channel[c].hshift != 0 || image.channel[c].vshift != 0) {
+      return false;
```

Comment:
See also https://github.com/libjxl/libjxl/pull/35

It's not incorrect to do palette on channels that have nonzero shift; this could in fact be useful, and the spec does not disallow it.

It is however incorrect to do palette on channels of different dimensions (which we already test elsewhere).

In our code we assume that when the dimensions are the same, the hshift/vshift is also the same, which I think is an invariant that should hold in all sane cases but there are some weird cases where it isn't necessarily the case: for example when you have a 3x100 3-channel image on which you do a 100-color palette and then do a 2-channel palette on the palette itself (which will also happen to be a 3x100 channel with hshift -1) plus the index image itself (which is a 3x100 channel with hshift 0).

So it is good to check that the hshift/vshifts are **the same** (so all equal to those of `begin_c`), because that's the actual invariant we assume to hold. In the reconstruction, the hshift/vshift of the channels [`begin_c`+1, `end_c`] will be restored to those of channel `begin_c`, so if this invariant is violated, we get a problematic case where MetaApply + undo_transforms does not produce something with the original channel metadata.

Reply:
#35 looks promising, but it does not fix the problem =(
Going to check if the invariant is held.

Reply:
PTAL